src/app/public/src/modules/table-of-contents/table-of-contents.component.ts

```diff
@@ -9,4 +12,50 @@ import { StacheNavLink } from '../nav';
 export class StacheTableOfContentsComponent {
   @Input()
   public routes: StacheNavLink[];
+  private viewTop: number;
+  private documentBottom: number;
+
+  constructor(
+    private windowRef: StacheWindowRef,
+    private omnibarService: StacheOmnibarAdapterService,
+    private anchorService: StachePageAnchorService
+  ) {
+    this.windowRef.scrollEvent.subscribe(() => {
+      this.updateRoutesOnScroll(this.anchorService.pageAnchors);
+    });
+  }
+
+  public updateRoutesOnScroll(routes: any[]) {
+    if (routes && routes.length) {
+      this.updateView(routes);
+    }
+  }
+
+  // Updates ng-class level trigger to show/hide blue border marking location on page
+  public updateView(routes: BehaviorSubject<StacheNavLink>[]) {
```

Comment:
There shouldn't be a BehaviorSubject array in this component now,  It should be expecting a standard array of StacheNavLinks